src/ui/public/chrome/directives/kbn_chrome.js

```diff
@@ -26,14 +27,24 @@ export default function (chrome, internals) {
       },
 
       controllerAs: 'chrome',
-      controller($scope, $rootScope, $location, $http) {
+      controller($scope, $rootScope, $location, $http, Private) {
+
+        let unhashStates;
+        try {
+          unhashStates = Private(UnhashStatesProvider);
+        } catch (err) {
+          // we need to work correctly if appState and globalState aren't available
+        }
 
         // are we showing the embedded version of the chrome?
         internals.setVisibleDefault(!$location.search().embed);
 
         // listen for route changes, propogate to tabs
         const onRouteChange = function () {
           let { href } = window.location;
+          if (unhashStates) {
```

Comment:
To avoid mutating a value, we could also do:

```javascript
const { href } = window.location;
const url = unhashStates ? unhashStates.inAbsUrl(href) : href;
internals.trackPossibleSubUrl(url);
```

I think it's a bit easier to follow because you don't need to hold two possible values of one variable in your head as you read the code.